samcli/lib/providers/sam_function_provider.py

```diff
@@ -277,32 +288,35 @@ def _build_function_configuration(
         samcli.commands.local.lib.provider.Function
             Function configuration
         """
+        metadata = resource_properties.get("Metadata", None)
+        if metadata and "DockerContext" in metadata:
+            metadata["DockerContext"] = SamLocalStackProvider.normalize_resource_path(stack, metadata["DockerContext"])
+
         return Function(
-            stack_path=stack_path,
+            stack_path=stack.stack_path,
             name=name,
             functionname=resource_properties.get("FunctionName", name),
             packagetype=resource_properties.get("PackageType", ZIP),
             runtime=resource_properties.get("Runtime"),
             memory=resource_properties.get("MemorySize"),
             timeout=resource_properties.get("Timeout"),
             handler=resource_properties.get("Handler"),
-            codeuri=codeuri,
+            codeuri=codeuri and SamLocalStackProvider.normalize_resource_path(stack, codeuri),
```

Comment:
```
SamLocalStackProvider.normalize_resource_path(stack, codeuri) if codeuri else None
```

would that be more descriptive? is that what you are going for?

Reply:
ok, makes sense, consistent with `imageuri=imageuri if imageuri else resource_properties.get("ImageUri"),`